tier4/Pilot.Auto
# ROS2 Porting: dummy_diag_publisher

# dummy_diag_publisher

## Summary

Convert the `dummy_diag_publisher` package to ROS2. This was relatively straightforward apart from the `dynamic_reconfigure` dependency.

Notable changes:
* Additional structs used to represent the original configuration generated using `dynamic_reconfigure`
* Remove fmt and boost deps since they are no longer used
* Add `rqt_reconfigure` to the `package.xml` as an `exec_depend`

## Testing

1. Enter ADE environment and build the workspace up to the package

```
colcon build --packages-up-to dummy_diag_publisher
. install/setup.bash
```

2. Launch files can be launch successfully (make sure that you run rosdep to get the `rqt_reconfigure` package:

```
ros2 launch dummy_diag_publisher dummy_diag_publisher.launch.xml launch_rqt:=true
```

## Discussion Points
* `rqt_runtime_monitor` has not been ported to ROS2, So visualization of these changes are difficult. Are there ROS2 alternatives? I couldn't find anything in my searches.
